### PR TITLE
docs/admin: clarify SecurityContextDeny and when it should be used

### PR DESCRIPTION
**Pull request by ericchiang**
This is based on my current understanding of the feature, which may be wrong. Needs review.

cc @kubernetes/sig-auth-pr-reviews  @gurvindersingh

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/3011)

### CODE REVIEW

File: docs/admin/admission-controllers.md [hunk 1]
```diff
@@ -197,7 +197,7 @@ We strongly recommend using this plug-in if you intend to make use of Kubernetes
 
 ### SecurityContextDeny
 
-This plug-in will deny any pod with a [SecurityContext](/docs/user-guide/security-context) that defines options that were not available on the `Container`.
+This plug-in will deny any pod that attempts to set certain escalating [SecurityContext](/docs/user-guide/security-context) fields. This should be enabled if a cluster doesn't utilize [pod security policies](/docs/user-guide/pod-security-policy) to restrict the set of values a security context can take.
```
Comment (gurvindersingh) [inline docs/admin/admission-controllers.md]: @ericchiang API server will block if SC is simply present irrespective of any value in it given this admission controller is enabled. So you may want to update the text in first line. Beyond that, I think the changes looks OK.

Comment (ericchiang) [inline docs/admin/admission-controllers.md]: From what I can tell based on the code[0], certain fields aren't limited e.g. FsGroup and RunAsNonRoot.[1]

[0] https://github.com/kubernetes/kubernetes/blob/v1.6.0-rc.1/plugin/pkg/admission/securitycontext/scdeny/admission.go
[1] https://github.com/kubernetes/kubernetes/blob/v1.6.0-rc.1/pkg/api/v1/types.go#L2381

Comment (gurvindersingh) [inline docs/admin/admission-controllers.md]: yup, you are correct. patch LGTM

**Comment by chenopis** [comment]
@gurvindersingh So is this Tech LGTM and good to go for merge?

**Comment by gurvindersingh** [comment]
@chenopis I think we can merge this. Info can be added about which fields are allowed in SC when `SecurityContextDeny` admission controller is enabled. But that can be added in a subsequent PR.

**Pull request merged**